File: ListView/app/src/main/java/com/example/listas/MainActivity.java
package com.example.listas;

import android.os.Bundle;
import android.view.View;
import android.widget.Adapter;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    private TextView t1;
    private ListView l1;
    String [] paises = {"España", "Francia", "Portugal"};
    String [] poblacion = {"40000000","50000000","35000000"};

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        t1 = findViewById(R.id.pais);
        l1 = findViewById(R.id.lista);

        ArrayAdapter<String> adaptador =
                new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1,paises);
        l1.setAdapter(adaptador);

        l1.setOnItemClickListener(new AdapterView.OnItemClickListener(){
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id){
                t1.setText("La población de " + l1.getItemAtPosition(position) + " es " +
                        poblacion[position]);
            }
        });
    }
}

File: RecyclerViewKotlin/app/src/main/java/com/example/recyclerviewkotlin/MainActivity.kt
package com.example.recyclerviewkotlin

import android.net.Uri
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {
    lateinit var rv: RecyclerView
    var rva: RecyclerView.Adapter<*>? = null
    var lm: RecyclerView.LayoutManager? = null
    var montes: MutableList<Monte> = ArrayList()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        // Leemos los datos
        readData()

        // Enlazamos la interfaz gráfica del RecyclerView con el código
        rv = findViewById(R.id.rv)
        // Establecemos que los elementos del RecyclerView se apilen verticalmente
        lm = LinearLayoutManager(this)
        // Creamos un adaptador para el RecyclerView
        rva = Adaptador(this, montes)
        // Enlazamos el adaptador con el objeto RecyclerView
        rv.setAdapter(rva)
        rv.setLayoutManager(lm)
    }

    // Añadimos una nueva montaña al RecyclerView
    fun addElemento(v: View?) {
        // Creamos el objeto de la nueva montaña
        val m = Monte(
            "Mont blanc", "Europa", 4855,
            R.drawable.mont, Uri.parse("https://en.wikipedia.org/wiki/Mont_Blanc")
        )
        // Lo añadimos a la lista de montes
        montes.add(m)
        // Notificamos al adaptador que hemos insertado una nueva montaña
        rva!!.notifyItemInserted(montes.size)
    }

    // Leemos los datos de las montañas del fichero arrays.xml
    fun readData() {
        val res = resources
        val nombres: Array<out String> = res.getStringArray(R.array.nombres)
        val alturas = res.getIntArray(R.array.alturas)
        val continentes: Array<out String> = res.getStringArray(R.array.continentes)
        val URLS: Array<out String> = res.getStringArray(R.array.URLS)
        val fotos = res.obtainTypedArray(R.array.fotos)
        val nMontes = nombres.size
        for (i in 0 until nMontes) {
            val nombre = nombres[i].toString()
            val altura = alturas[i]
            val continente = continentes[i].toString()
            val url = Uri.parse(URLS[i].toString())
            val foto = fotos.getResourceId(i, R.drawable.mul)
            val m = Monte(nombre, continente, altura, foto, url)
            montes.add(m)
        }
    }
}